### PR TITLE
Add city completion script

### LINKED ISSUE #251: Command line completion for `city` command
The new `city` command is used to run all completed cities. This could benefit from command line completion assistance.

### PR DESCRIPTION
**Pull request by gonzaponte**
This PR adds CLI completion to the city command. Examples:

```
city <tab>
# will display
# beersheba    buffy        diomira      esmeralda    hypathia     isaura       penthesilea  trude      
# berenice     detsim       dorothea     eutropia     irene        isidora      phyllis 
```
```
city e<tab>
# will display
# esmeralda eutropia
```
etc.

Closes #251

### CODE REVIEW

File: bin/city-completion [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    if [[ $cityname == *"test"* ]]; then continue; fi
+    if [[ $cityname == *"components"* ]]; then continue; fi
+    if [[ $cityname == "__"* ]]; then continue; fi
+    if [[ $cityname == *"~" ]]; then continue; fi
+    cityname=${cityname/.py/}
+    COMPREPLY+=("$cityname")
+  done
+}
+
+#complete -W "aaa bbb ccc dddd" city
```
Comment (jacg) [inline bin/city-completion]: Get rid of this obsolete comment.

**Comment by jwaiton** [comment]
The code looks good, removes all the non-relevant cities and doesn't perform inappropriately when misused.

One issue I've noted is that the expected behaviour of showing available cities based on the typed characters doesn't work as desired.

Expected:
```
city e<tab>
# will display
# esmeralda eutropia
```
Output:
```
city e<tab>
# will display
# beersheba detsim esmeralda irene penthesilea trude ...
```

Reading the bash, this feature isn't included, is this still desired as a feature?

**Comment by gonzaponte** [comment]
Hmm, it worked for me and I assumed it was some black magic I wasn't aware of. I've made a change to include completion for partial input. Please check.

**Comment by jwaiton** [comment]
It may be based on the shell I'm using, bash 5.1.16. I think zsh has differing autocomplete behaviour.

The updated change works, and the behaviour seems appropriate and convenient. Good work :)

**Review by jwaiton — APPROVED** [review]
The CLI `city` input now has completion capabilities, offering all relevant cities to the user upon use of <tab><tab> based on their input.

The code works as expected, and I haven't found any weird edge cases. Good job!

**Pull request merged**